Ruby/Ruby.sublime-syntax

```diff
@@ -203,8 +203,70 @@ contexts:
       scope: constant.language.ruby
     - match: '\b(__(FILE|LINE|ENCODING)__|self)\b(?![?!])'
       scope: variable.language.ruby
-    - match: '\b(0[xX]\h(_?\h)*|\d(_?\d)*(\.(?![^[:space:][:digit:]])(_?\d)*)?([eE][-+]?\d(_?\d)*)?|0[bB][01]+)\b'
-      scope: constant.numeric.ruby
+    # numbers with 'ir' suffix are invalid
+    - match: '\b(?:\d(?:_?\d)*(?:\.\d(?:_?\d)*)?(?:[eE][-+]?\d(?:_?\d)*)?|0[dDxXoObB]?\h(?:_?\h)*)ir\b'
+      scope: invalid.illegal.ruby
+    - match: '\b(0[dD])\d(?:_?\d)*(r)?(i)?\b'
+      scope: constant.numeric.integer.decimal.ruby
+      captures:
+        1: punctuation.definition.numeric.decimal.ruby
+        2: storage.type.numeric.rational.ruby
+        3: storage.type.numeric.imaginary.ruby
+    - match: '\b(0[xX])\h(?:_?\h)*(r)?(i)?\b'
+      scope: constant.numeric.integer.hexadecimal.ruby
+      captures:
+        1: punctuation.definition.numeric.hexadecimal.ruby
+        2: storage.type.numeric.rational.ruby
+        3: storage.type.numeric.imaginary.ruby
+    - match: '\b(0[oO]?)[0-7](?:_?[0-7])*(r)?(i)?\b'
+      scope: constant.numeric.integer.octal.ruby
+      captures:
+        1: punctuation.definition.numeric.octal.ruby
+        2: storage.type.numeric.rational.ruby
+        3: storage.type.numeric.imaginary.ruby
+    - match: '\b(0[bB])[01](?:_?[01])*(r)?(i)?\b'
+      scope: constant.numeric.integer.binary.ruby
+      captures:
+        1: punctuation.definition.numeric.binary.ruby
+        2: storage.type.numeric.rational.ruby
+        3: storage.type.numeric.imaginary.ruby
+    # 1.1e1i, 1e1i
+    - match: '\b\d(?:_?\d)*(?:(\.)\d(?:_?\d)*)?[eE][-+]?\d(?:_?\d)*(i)\b'
+      scope: constant.numeric.complex.imaginary.ruby
+      captures:
+        1: punctuation.separator.decimal.ruby
+        2: storage.type.numeric.imaginary.ruby
+    # 1.1ri, 1.1i, 1ri, 1i
+    - match: '\b\d(?:_?\d)*(?:(\.)\d(?:_?\d)*)?(r)?(i)\b'
+      scope: constant.numeric.complex.imaginary.ruby
+      captures:
+        1: punctuation.separator.decimal.ruby
+        2: storage.type.numeric.rational.ruby
+        3: storage.type.numeric.imaginary.ruby
+    # 1.i
+    - match: '\b\d(?:_?\d)*(\.)(i)\b'
+      scope: constant.numeric.complex.imaginary.ruby
+      captures:
+        1: punctuation.separator.decimal.ruby
+        2: storage.type.numeric.imaginary.ruby
+    # 1.1r, 1r
+    - match: '\b\d(?:_?\d)*(?:(\.)\d(?:_?\d)*)?(r)\b'
+      scope: constant.numeric.rational.ruby
+      captures:
+        1: punctuation.separator.decimal.ruby
+        2: storage.type.numeric.rational.ruby
+    # 1.1e1, 1e1
+    - match: '\b\d(?:_?\d)*(?:(\.)\d(?:_?\d)*)?[eE][-+]?\d(?:_?\d)*\b'
+      scope: constant.numeric.float.ruby
+      captures:
+        1: punctuation.separator.decimal.ruby
+    # 1.1, 1.
+    - match: '\b\d(?:_?\d)*(\.)(?:\d(?:_?\d)*\b)?'
+      scope: constant.numeric.float.ruby
+      captures:
+        1: punctuation.separator.decimal.ruby
+    - match: '\b\d(?:_?\d)*\b'
+      scope: constant.numeric.integer.ruby
```

Comment:
If we decide to merge the `rational` numbers rule into float/integer, here is a suggestion to do so without introducing additional rules:

```yaml
    # 1.1r, 1r, 1.1, 1.
    - match: '\b\d(?:_?\d)*(\.)(?:(?:\d(?:_?\d)*)?(r)?\b)?'
      scope: constant.numeric.float.ruby
      captures:
        1: punctuation.separator.decimal.ruby
        2: storage.type.numeric.rational.ruby
    # 1, 1r, 1_1, 1_1r
    - match: '\b\d(?:_?\d)*(r)?\b'
      scope: constant.numeric.integer.ruby
      captures:
        1: storage.type.numeric.rational.ruby
```